diff --git a/tracktable/Core/CMakeLists.txt b/tracktable/Core/CMakeLists.txt
index 9ae861c3..f4bdec22 100644
--- a/tracktable/Core/CMakeLists.txt
+++ b/tracktable/Core/CMakeLists.txt
@@ -264,3 +264,14 @@ install(
   LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
   ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
 )
+
+if (TRACKTABLE_PYTHON)
+  message(STATUS "Installing C++ shared libraries alongside Python extensions.")
+  install(
+    TARGETS TracktableCore
+    RUNTIME DESTINATION "${CMAKE_INSTALL_PREFIX}/Python/tracktable/lib"
+    LIBRARY DESTINATION "${CMAKE_INSTALL_PREFIX}/Python/tracktable/lib"
+    RUNTIME DESTINATION "${CMAKE_INSTALL_PREFIX}/Python/tracktable/lib"
+    ARCHIVE
+  )
+endif ()
\ No newline at end of file
diff --git a/tracktable/Domain/CMakeLists.txt b/tracktable/Domain/CMakeLists.txt
index d8d9e989..7ede9636 100644
--- a/tracktable/Domain/CMakeLists.txt
+++ b/tracktable/Domain/CMakeLists.txt
@@ -93,6 +93,16 @@ install(
   ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
 )
 
+
+if (TRACKTABLE_PYTHON)
+  install(
+    TARGETS TracktableDomain
+    RUNTIME DESTINATION "${CMAKE_INSTALL_PREFIX}/Python/tracktable/lib"
+    LIBRARY DESTINATION "${CMAKE_INSTALL_PREFIX}/Python/tracktable/lib"
+    RUNTIME DESTINATION "${CMAKE_INSTALL_PREFIX}/Python/tracktable/lib"
+    ARCHIVE
+  )
+endif ()
 install(
   FILES ${Domain_HEADERS}
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${TRACKTABLE_INCLUDEDIR}/Domain
diff --git a/tracktable/PythonWrapping/CMakeLists.txt b/tracktable/PythonWrapping/CMakeLists.txt
index f40a1040..5b0193cc 100644
--- a/tracktable/PythonWrapping/CMakeLists.txt
+++ b/tracktable/PythonWrapping/CMakeLists.txt
@@ -149,6 +149,15 @@ link_directories(
   ${Tracktable_BINARY_DIR}/lib
 )
 
+if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+  message(STATUS "Setting RPATH for dynamic libraries to look in @loader_path/../lib.")
+  set(CMAKE_OSX_RPATH ON)
+  list(APPEND CMAKE_INSTALL_RPATH "@loader_path/../lib")
+elseif (CMAKE_SYSTEM_NAME STREQUAL "Linux")
+  message(STATUS "Setting RPATH for dynamic libraries to look in \$ORIGIN/../lib.")
+  list(APPEND CMAKE_INSTALL_RPATH "\$ORIGIN/../lib")
+endif ()
+
 set(Tracktable_PYTHON_DIR ${Tracktable_SOURCE_DIR}/tracktable/Python/tracktable)
 
 # We use Python-like naming for these files for two reasons. First,
